app/controllers/instrument_duration_rates_controller.rb

```diff
@@ -59,4 +62,29 @@ def set_product_duration_rates
 
     @product_duration_rates = @product_duration_rates.sort_by { |dr| dr.min_duration || 1_000 }
   end
+
+  def log_changes(previous_duration_rates)
+    previous_duration_rates.sort_by! { |dr| dr.min_duration || 1_000 }
+    new_duration_rates = @product.duration_rates.sort_by { |dr| dr.min_duration || 1_000 }
+
+    if previous_duration_rates.length != @product.duration_rates.length
+      LogEvent.log(@product, :duration_rates_change, current_user, metadata: { before: build_log_metadata(previous_duration_rates), after: build_log_metadata(new_duration_rates) })
+      return
+    end
+
+    changes = false
+
+    previous_duration_rates.zip(@product.duration_rates).each do |previous_duration_rate, new_duration_rate|
+      unless previous_duration_rate.min_duration == new_duration_rate.min_duration && previous_duration_rate.rate == new_duration_rate.rate
+        changes = true
+        break
+      end
+    end
```

Comment:
What if you use `map` here instead:
```suggestion
    changes = previous_duration_rates.zip(@product.duration_rates).map do |previous_duration_rate, new_duration_rate|
     previous_duration_rate.min_duration != new_duration_rate.min_duration || previous_duration_rate.rate != new_duration_rate.rate
    end
```
then...
```
 LogEvent.log(...) if changes.any?
```
https://apidock.com/ruby/Enumerable/any%3F
Reads a little clearer to me, what do you think?